dataflow/pkg/db/drivers/mongo/mongo.go

```diff
@@ -218,6 +218,20 @@ func (ad *adapter) UnlockBucketLifecycleSched(bucketName string) int {
 	return unlock(ss, bucketName)
 }
 
+func (ad *adapter) LockRestoreObjectSched(ObjectKey string) int {
+	ss := ad.s.Copy()
+	defer ss.Close()
+
+	return lock(ss, ObjectKey, 300) //One schedule is supposed to be finished in 300 seconds
```

Comment:
so the call to this outer level API will block for >=300 seconds ? it will timeout, please explain this lock/unlock and 300 seconds logic